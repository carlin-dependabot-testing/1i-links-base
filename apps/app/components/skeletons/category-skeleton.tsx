import { SectionHeader } from '@/components/layout/section-header'

import { Skeleton } from '@links-base/ui/skeleton'

export function CategorySkeleton() {
  return (
    <div className="space-y-6">
      <SectionHeader title="Category" description="Category">
        <Skeleton className="h-8 w-48" />
      </SectionHeader>

      <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-3">
        {Array.from({ length: 6 }).map((_, i) => (
          <Skeleton key={i} className="h-[200px] rounded-lg" />
        ))}
      </div>
    </div>
  )
}
