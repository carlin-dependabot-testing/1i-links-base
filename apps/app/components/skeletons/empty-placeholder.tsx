import * as React from 'react'

import { cn } from '@links-base/ui/utils'

export function EmptyPlaceholder({
  className,
  children,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) {
  return (
    <div
      className={cn(
        'flex min-h-[400px] flex-col items-center justify-center rounded-md border border-dashed p-8 text-center animate-in fade-in-50',
        className
      )}
      {...props}
    >
      <div className="mx-auto flex max-w-[420px] flex-col items-center justify-center text-center">
        {children}
      </div>
    </div>
  )
}

EmptyPlaceholder.Icon = function EmptyPlaceHolderIcon({
  className,
  ...props
}: React.ComponentProps<'div'>) {
  return (
    <div
      className={cn(
        'flex h-20 w-20 items-center justify-center rounded-full bg-muted',
        className
      )}
      {...props}
    />
  )
}

EmptyPlaceholder.Title = function EmptyPlaceholderTitle({
  className,
  ...props
}: React.ComponentProps<'h2'>) {
  return (
    <h2 className={cn('mt-6 text-xl font-semibold', className)} {...props} />
  )
}

EmptyPlaceholder.Description = function EmptyPlaceholderDescription({
  className,
  ...props
}: React.ComponentProps<'p'>) {
  return (
    <p
      className={cn(
        'mb-8 mt-2 text-center text-sm font-normal leading-6 text-muted-foreground',
        className
      )}
      {...props}
    />
  )
}
