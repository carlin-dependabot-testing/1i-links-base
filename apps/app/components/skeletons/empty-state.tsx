import { type LucideIcon } from 'lucide-react'

import { Button } from '@links-base/ui/button'

export type EmptyStateProps = {
  /** The title to display */
  title: string
  /** Optional description text */
  description?: string
  /** Optional icon to display */
  icon?: LucideIcon
  /** Optional action button configuration */
  action?:
    | {
        label: string
        onClick: () => void
      }
    | React.ReactNode
  /** Optional custom CSS classes */
  className?: string
  /** Optional height of the empty state */
  height?: string
}

/**
 * A reusable empty state component to display when there is no data
 */
export const EmptyState = ({
  title,
  description,
  icon: Icon,
  action,
  className = '',
  height = '400px'
}: EmptyStateProps) => {
  return (
    <div
      className={`flex min-h-[${height}] flex-col items-center justify-center rounded-lg border border-dashed border-gray-300 bg-gray-50/50 px-5 py-5 text-center dark:border-empty-border dark:bg-empty ${className}`}
      role="status"
      aria-label={title}
    >
      {Icon && (
        <Icon
          className="mb-3 h-10 w-10 text-gray-400 dark:text-empty-icon"
          aria-hidden="true"
        />
      )}
      <h3 className="mb-2 text-lg font-medium text-gray-900 dark:text-empty-title">
        {title}
      </h3>
      {description && (
        <p className="mb-3 text-sm text-gray-500 dark:text-empty-text">
          {description}
        </p>
      )}
      {action &&
        (typeof action === 'object' && 'label' in action ? (
          <Button onClick={action.onClick} variant="outline" size="sm">
            {action.label}
          </Button>
        ) : (
          action
        ))}
    </div>
  )
}
