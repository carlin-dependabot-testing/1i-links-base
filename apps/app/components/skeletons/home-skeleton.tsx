import { Skeleton } from '@links-base/ui/skeleton'

export const HomeSkeleton = () => {
  return (
    <div data-testid="home-skeleton" className="p-6 md:p-10">
      {/* Header skeleton */}
      <div className="mb-6 flex items-center justify-between">
        <Skeleton className="h-9 w-24" />
        <Skeleton className="h-10 w-[180px]" />
      </div>

      {/* Pinned Links skeleton */}
      <div className="mb-8">
        <Skeleton className="mb-4 h-6 w-24" /> {/* "Pinned Links" text */}
        <div className="flex gap-4 overflow-x-auto pb-2">
          {[1, 2, 3, 4, 5].map(i => (
            <div key={i} className="flex-shrink-0">
              <Skeleton className="h-[100px] w-[200px] rounded-lg border bg-card shadow-sm" />
            </div>
          ))}
        </div>
      </div>

      {/* Search and filters skeleton */}
      <div className="mb-6 flex items-center gap-4">
        <Skeleton className="h-10 w-[200px]" /> {/* Search input */}
        <Skeleton className="h-10 w-[150px]" /> {/* Filter button */}
      </div>

      {/* Categories and Links */}
      <div className="space-y-8">
        {/* Development section */}
        <div>
          <div className="mb-4 flex items-center gap-2">
            <Skeleton className="h-5 w-5" /> {/* Icon */}
            <Skeleton className="h-6 w-32" /> {/* Category name */}
            <Skeleton className="h-5 w-16" /> {/* Link count */}
          </div>

          <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-6">
            {[1, 2, 3, 4, 5, 6].map(i => (
              <Skeleton
                key={i}
                className="h-[140px] rounded-lg border bg-card/50 shadow-sm"
              />
            ))}
          </div>
        </div>

        {/* Communication section */}
        <div>
          <div className="mb-4 flex items-center gap-2">
            <Skeleton className="h-5 w-5" /> {/* Icon */}
            <Skeleton className="h-6 w-36" /> {/* Category name */}
            <Skeleton className="h-5 w-16" /> {/* Link count */}
          </div>

          <div className="grid gap-4 md:grid-cols-2 lg:grid-cols-6">
            {[1, 2, 3].map(i => (
              <Skeleton
                key={i}
                className="h-[140px] rounded-lg border bg-card/50 shadow-sm"
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}
