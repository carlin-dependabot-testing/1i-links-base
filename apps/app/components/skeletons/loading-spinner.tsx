import { cn } from '@links-base/ui/utils'

export type LoadingSpinnerProps = {
  size?: 'sm' | 'md' | 'lg'
  className?: string
}

/**
 * A loading spinner component that provides visual feedback during loading states
 */
export const LoadingSpinner = ({
  size = 'md',
  className
}: LoadingSpinnerProps) => {
  const sizeClasses = {
    sm: 'w-4 h-4',
    md: 'w-8 h-8',
    lg: 'w-12 h-12'
  }

  return (
    <div
      role="status"
      className="flex h-full min-h-[100px] w-full items-center justify-center"
      data-testid="loading-spinner"
    >
      <div
        className={cn(
          'animate-spin rounded-full border-4 border-primary border-r-transparent',
          sizeClasses[size],
          className
        )}
      >
        <span className="sr-only">Loading...</span>
      </div>
    </div>
  )
}
