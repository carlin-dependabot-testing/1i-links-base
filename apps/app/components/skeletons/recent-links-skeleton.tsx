import { LINKS_GRID_CLASSES } from '@/constants'

import { Skeleton } from '@links-base/ui/skeleton'

export const RecentLinksSkeleton = () => {
  return (
    <div className="flex flex-col gap-8">
      {/* Header with button skeleton */}
      <div className="mb-4 flex items-center justify-between">
        <div>
          <Skeleton className="mb-2 h-9 w-[250px]" /> {/* Title */}
          <Skeleton className="h-5 w-[200px]" /> {/* Subtitle */}
        </div>
        <Skeleton className="h-10 w-[120px]" /> {/* Request Link button */}
      </div>

      {/* First date group */}
      <div>
        <div className="mb-2">
          <Skeleton className="mb-1 h-7 w-48" /> {/* Date heading */}
          <Skeleton className="mb-4 h-5 w-24" /> {/* "X links added" text */}
        </div>
        <div className={LINKS_GRID_CLASSES}>
          {[1, 2].map(i => (
            <Skeleton
              key={i}
              className="h-[140px] rounded-lg border bg-card/50 shadow-sm"
            />
          ))}
        </div>
      </div>

      {/* Second date group */}
      <div>
        <div className="mb-2">
          <Skeleton className="mb-1 h-7 w-48" /> {/* Date heading */}
          <Skeleton className="mb-4 h-5 w-24" /> {/* "X links added" text */}
        </div>
        <div className={LINKS_GRID_CLASSES}>
          {[1, 2, 3, 4].map(i => (
            <Skeleton
              key={i}
              className="h-[140px] rounded-lg border bg-card/50 shadow-sm"
            />
          ))}
        </div>
      </div>
    </div>
  )
}
