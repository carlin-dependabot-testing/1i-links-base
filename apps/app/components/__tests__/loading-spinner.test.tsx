import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'

import { LoadingSpinner } from '@/components/skeletons/loading-spinner'

const renderComponent = (props = {}) => {
  return render(<LoadingSpinner {...props} />)
}

describe('<LoadingSpinner />', () => {
  it('<LoadingSpinner /> should render correctly', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should be accessible', async () => {
    const { container } = renderComponent()
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })

  it('should render with different sizes', () => {
    const { rerender } = renderComponent({ size: 'sm' })
    expect(screen.getByRole('status')).toBeInTheDocument()

    rerender(<LoadingSpinner size="lg" />)
    expect(screen.getByRole('status')).toBeInTheDocument()
  })
})
