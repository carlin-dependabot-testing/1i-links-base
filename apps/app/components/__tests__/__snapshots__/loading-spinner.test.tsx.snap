// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> <LoadingSpinner /> should render correctly 1`] = `
<div>
  <div
    class="flex h-full min-h-[100px] w-full items-center justify-center"
    data-testid="loading-spinner"
    role="status"
  >
    <div
      class="animate-spin rounded-full border-4 border-primary border-r-transparent w-8 h-8"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
